service/controller/v3/resource/instance/create.go

```diff
@@ -33,187 +31,205 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	{
-		d, err := deploymentsClient.Get(ctx, key.ClusterID(customObject), vmssDeploymentName)
-		if IsDeploymentNotFound(err) {
-			// fall through
+	if !resourceStatusExists(customObject, "Stage") {
```

Comment:
The diff is shite. There are 3 stages the `instance` resource ensures sequentially. These stages where executed in a rather uncontrolled way which is the reason for the issue this PR addresses and fixes. These are the three stages. 

- Uploading ARM templates, described by stage `DeploymentInitialized`.
- Waiting for ARM templates to be applied, described by stage `ProvisioningSuccessful`.
- Draining and upgrading the VMSS instances, described by stage `InstancesUpgrading`.

The stages are executed one after another and the instance cycles through them reliably until all necessary upgrade magic is sorted.

Reply:
Thanks for the explanation that makes sense.

Reply:
Should I add it to the godoc for `EnsureCreated`?

Reply:
Yes that would help

Reply:
+1 that would be useful.

Reply:
I added godoc.